### PR TITLE
Remove black formatter to use only ruff

### PR DESCRIPTION
**Pull request by Wauplin**
Following https://astral.sh/blog/the-ruff-formatter, I suggest we remove `black` formatter completely. It was already not very used anyway. Made a few changes so that current code is formatted the same with both `black` and `ruff` to reduce inconveniences for existing contributors.
N°1 argument IMO is speed. N°2 is that it's convenient to have everything handled by the same tool.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Great, thank you! Would be great to align with transformers/accelerate/diffusers/others

cc @charliermarsh, you might be interested :)

**Comment by Wauplin** [comment]
Thanks @LysandreJik for the review :)

And immensely grateful @charliermarsh for the amazing work on Ruff :heart:

**Comment by charliermarsh** [comment]
Amazing, thanks @Wauplin and @LysandreJik! So cool to see :)

File: utils/_legacy_check_future_compatible_signatures.py [hunk 1]
```diff
@@ -173,17 +172,14 @@ def generate_hf_api_module() -> str:
...
     """
-    Format some code with black+ruff. Cannot be done "on the fly" so we first save the code in a temporary file.
+    Format some code with ruff. Cannot be done "on the fly" so we first save the code in a temporary file.
     """
-    # Format with black
-    code = black.format_file_contents(code, fast=False, mode=black.FileMode(line_length=119))
-
     # Format with ruff
     with tempfile.TemporaryDirectory() as tmpdir:
         filepath = Path(tmpdir) / "__init__.py"
```
Comment (ArthurZucker) [inline utils/_legacy_check_future_compatible_signatures.py:185]: The call to `ruff` without arguments will not format but check. 
See #27144 where we heavily rely on the formatting of the `# Copied from`

Comment (Wauplin) [reply utils/_legacy_check_future_compatible_signatures.py:185]: Good catch!
Fixed this in https://github.com/huggingface/huggingface_hub/pull/1824.

**Pull request merged**